### PR TITLE
feat: add --ignore-labels flag

### PR DESCRIPTION
**Pull request by ebezzi**
Add an `--ignore-labels` flag. If true, the validator will not check for reserved column names that would be added by the `--add-labels` option.

This is useful if you want to validate a dataset that has already been uploaded to the cellxgene data portal.

### CODE REVIEW

File: cellxgene_schema_cli/tests/fixtures/examples_validate.py [hunk 1]
```diff
@@ -203,7 +203,7 @@
 adata_with_labels = anndata.AnnData(
     X=sparse.csr_matrix(X),
     obs=pd.concat([good_obs, obs_expected], axis=1),
-    var=pd.concat([good_var, var_expected], axis=1),
+    var=var_expected,
```
Comment (ebezzi) [inline cellxgene_schema_cli/tests/fixtures/examples_validate.py:206]: Using `pd.concat` here corrupts the resulting dataframe which leads to a failed validation. `var_expected` is a superset of `good_var` so this is also fine.

File: cellxgene_schema_cli/tests/test_validate.py [hunk 2]
```diff
@@ -204,6 +204,32 @@ def test_get_dictionary_mapping_curie(self):
...
+        is_valid = validator.validate_adata()
+
+        self.assertFalse(is_valid)
+
+    def test_validating_labeled_h5ad_should_pass_if_flag_set(self):
+
+        validator = Validator(ignore_labels=True)
+        import copy
+        validator.adata = copy.deepcopy(self.test_adata_with_labels)
+        validator.adata.uns["X_normalization"] = 'none'
```
Comment (ebezzi) [inline cellxgene_schema_cli/tests/test_validate.py:227]: Won't pass validation if `X_normalization` isn't `none`.

Comment (brianraymor) [reply cellxgene_schema_cli/tests/test_validate.py:226]: Also remember that we're deprecating `X_normalization` in 3.0.0.

Comment (ebezzi) [reply cellxgene_schema_cli/tests/test_validate.py:226]: Good point. I feel even less guilty about this hack now.

File: cellxgene_schema_cli/tests/fixtures/examples_validate.py (hunk 1 above)
Comment (MDunitz) [reply cellxgene_schema_cli/tests/fixtures/examples_validate.py:206]: just curious -- why isnt that also an issue for obs?

File: cellxgene_schema_cli/tests/test_validate.py [hunk 3]
```diff
@@ -204,6 +203,32 @@ def test_get_dictionary_mapping_curie(self):
...
+        validator = Validator()
+        validator.adata = self.test_adata_with_labels
+        is_valid = validator.validate_adata()
+
+        self.assertFalse(is_valid)
+
+    def test_validating_labeled_h5ad_should_pass_if_flag_set(self):
+
+        validator = Validator(ignore_labels=True)
+        import copy
```
Comment (MDunitz) [inline cellxgene_schema_cli/tests/test_validate.py:224]: is there a reason for putting the import here?

File: cellxgene_schema_cli/tests/fixtures/examples_validate.py (hunk 1 above)
Comment (ebezzi) [reply cellxgene_schema_cli/tests/fixtures/examples_validate.py:206]: I think the issue is that duplicate columns are merged in a weird way (if you select the column, you get a `Dataframe` and not a `Series`). The two obs do not have duplicate columns.

File: cellxgene_schema_cli/tests/test_validate.py (hunk 3 above)
Comment (ebezzi) [reply cellxgene_schema_cli/tests/test_validate.py:224]: Not necessarily. I will move it to the top.

File: cellxgene_schema_cli/cellxgene_schema/cli.py [hunk 4]
```diff
@@ -47,7 +53,7 @@ def schema_validate(h5ad_file, add_labels_file, verbose):
     print("Loading validator modules")
     from .validate import validate
 
-    is_valid, _, _ = validate(h5ad_file, add_labels_file, verbose=verbose)
+    is_valid, _, _ = validate(h5ad_file, add_labels_file, ignore_labels, verbose=verbose)
```
Comment (pablo-gar) [inline cellxgene_schema_cli/cellxgene_schema/cli.py:56]: Shouldn't it be? 
```suggestion
    is_valid, _, _ = validate(h5ad_file, add_labels_file, ignore_labels=ignore_labels, verbose=verbose)
```

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 5]
```diff
@@ -1312,7 +1318,7 @@ def validate(
         logging.basicConfig(level=logging.DEBUG)
     else:
         logging.basicConfig(level=logging.INFO, format="%(message)s")
-    validator = Validator()
+    validator = Validator(ignore_labels)
```
Comment (pablo-gar) [inline cellxgene_schema_cli/cellxgene_schema/validate.py:1321]:
```suggestion
    validator = Validator(ignore_labels=ignore_labels)
```

**Review by pablo-gar — CHANGES_REQUESTED** [review]
minor changes

File: cellxgene_schema_cli/cellxgene_schema/cli.py (hunk 4 above)
Comment (pablo-gar) [reply cellxgene_schema_cli/cellxgene_schema/cli.py:56]: better to explicitly name them in case more args get added in the future

**Review by pablo-gar — APPROVED** [review]

**Pull request merged**